### ISSUE TITLE
Icon fetch in Gecko/Blink uses document charset for URL encoding

### ISSUE THREAD

**Issue by saschanaz** [whatwg/notifications#209]
### What is the issue with the Notifications API Standard?

Part of #208 and #5.

Just found that Both Firefox and Chrome depends on [the legacy URL encoding behavior](https://html.spec.whatwg.org/multipage/urls-and-fetching.html#encoding-parsing-a-url), while [Safari has no icon support](https://searchfox.org/wubkat/rev/f8504f25d2ccc6a64957e53ab8c4e42885007fb0/Source/WebCore/Modules/notifications/NotificationResourcesLoader.cpp#52-56).

We can spec that as is, but maybe it's worth changing to use utf-8? For now I'll gather use counter for this.

**Comment by annevk** [comment]
Using UTF-8 would definitely be preferable. Thanks for adding a counter!

cc @beverloo

**Comment by saschanaz** [comment]
Non-UTF8 usage is [absolute zero](https://glam.telemetry.mozilla.org/fog/probe/web_notification_icon_url_encoding/explore?activeBuckets=%5B%22either_way%22%2C%22utf8%22%2C%22document_charset%22%2C%22neither_way%22%5D&aggType=avg&app_id=release&ping_type=*&timeHorizon=ALL). (It's all "either way", meaning the URL decoding result for icon is always same regardless of whether it's by UTF8 or by the document charset.)

We'll start making it always UTF8.

**Comment by saschanaz** [comment]
Spoke too soon, I found an error in the telemetry. Surely it couldn't be absolute zero...

**Comment by saschanaz** [comment]
The updated telemetry still reports absolute zero use of non-UTF8 URLs. We'll try dropping document charset use.

**Comment by annevk** [comment]
I realized this is already required by the specification (we don't parse URLs with encoding arguments) and it's tested too: https://github.com/web-platform-tests/wpt/pull/59775
